companion/lib/Service/Satellite.js

```diff
@@ -292,31 +300,41 @@ class ServiceSatellite extends ServiceBase {
 			socket.write(`KEY-PRESS ERROR MESSAGE="Missing DEVICEID"\n`)
 			return
 		}
-		if (!params.KEY) {
-			socket.write(`KEY-PRESS ERROR DEVICEID="${params.DEVICEID}" MESSAGE="Missing KEY"\n`)
+		const id = `${params.DEVICEID}`
+		const device = this.#devices.get(id)
+		if (device === undefined || device.socket !== socket) {
+			socket.write(`KEY-PRESS ERROR DEVICEID="${id}" MESSAGE="Device not found"\n`)
 			return
 		}
-		if (!params.PRESSED) {
-			socket.write(`KEY-PRESS ERROR DEVICEID="${params.DEVICEID}" MESSAGE="Missing PRESSED"\n`)
+		if (!params.KEY) {
+			socket.write(`KEY-PRESS ERROR DEVICEID="${id}" MESSAGE="Missing KEY"\n`)
 			return
 		}
-
-		const key = Number(params.KEY)
-		if (isNaN(key) || key > LEGACY_MAX_BUTTONS || key < 0) {
-			socket.write(`KEY-PRESS ERROR DEVICEID="${params.DEVICEID}" MESSAGE="Invalid KEY"\n`)
+		if (!params.PRESSED) {
+			socket.write(`KEY-PRESS ERROR DEVICEID="${id}" MESSAGE="Missing PRESSED"\n`)
 			return
 		}
-
 		const pressed = !isFalsey(params.PRESSED)
 
-		const id = `${params.DEVICEID}`
-		const device = this.#devices.get(id)
+		const keynum = Number(params.KEY)
+		const keyParse = params.KEY.toString().match(/^\+?(\d+)\/\+?(\d+)$/)
 
-		if (device && device.socket === socket) {
-			device.device.doButton(key, pressed)
+		if (
+			Array.isArray(keyParse) &&
+			Number(keyParse[1]) < device.device.gridSize.rows &&
+			Number(keyParse[2]) < device.device.gridSize.columns
+		) {
+			this.emit('click', Number(keyParse[2]), Number(keyParse[1]), pressed)
```

Comment:
I think these emits won't work, there isn't anything listening to them on this class. 
It would work if they were made `device.device.emit(....`, but another comment should negate the need to do this